closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -367,6 +373,54 @@ private static void checkFIPSCompatibility(Properties props) {
         }
     }
 
+    /**
+     * Check whether security property can be set.
+     *
+     * A security property that is set by a RestrictedSecurity profile,
+     * while FIPS security mode is enabled, cannot be reset programmatically.
+     *
+     * Every time an attempt to set a security property is made, a check is
+     * performed. If the above scenario holds true, a SecurityException is
+     * thrown.
+     *
+     * One can override this behaviour and allow the user to set any security
+     * property through the use of `-Dsemeru.fips.allowsetproperties=true'.
+     *
+     * @param key the security property that the user wants to set
+     * @throws SecurityException
+     *         if the security property is set by the profile and cannot
+     *         be altered.
+     */
+    public static void checkSetSecurityProperty(String key) {
+        if (debug != null) {
+            debug.println("RestrictedSecurity: Checking whether security property '"
+                    + key + "' can be set.");
+        }
+
+        /*
+         * Only disallow setting of security properties that are set by the active profile,
+         * if FIPS has been enabled.
+         *
+         * Allow any change, if the 'semeru.fips.allowsetproperties' flag is set to true.
+         */
+        if (userEnabledFIPS && !allowSetSecurityProps && alteredProperties.contains(key)) {
+            if (debug != null) {
+                debug.println("RestrictedSecurity: Security property '"
+                        + key + "' cannot be set.");
+                debug.println("If you want to override the check and allow all security"
+                        + "properties to be set, use `-Dsemeru.fips.allowsetproperties=true'.");
+                debug.println("BEWARE: You might not be FIPS compliant if you select to override!");
```

Comment:
Suggest shortening "security property" to just "property" ("security" seem redundant immediately after "RestrictedSecurity"). Please avoid mismatched quotes (back-tick and `'`).

Reply:
Shortened and fixed.